Updating read_scrape to match dates

erika-tyagi opened the pull request:
Updates to `read_scrape_data` to: 
* Fixes the date mismatch issues (i.e. where we have `Residents.Population` data on 1/1/21 but run the scrapers on 1/2/21 and want to forward fill population data within a time window for a facility). 
* Drops rows where all COVID variables are NA (e.g. where we only have population data) 

A few things to note: 
* Both of these are parameters (`coalesce_pop` and `drop_na_covid`) that default to TRUE. Is that what we want? Are the names descriptive enough? Thought about renaming `coalesce` to `coalesce_facilities` but decided that might break things. 
* I'm using the same window parameter for the `all_dates = FALSE` forward-filling as historical population forward-filling. This feels okay to me, I think? 
* I decided to handle the date window here (as opposed to create a new version of `last_not_na` because it (1) felt like a cleaner solution (otherwise we'd have to pass around the Date vector and the Facility.ID vector to the function, which got messy with groupings, and (2) was MUCH faster (adding another looping nest over df rows was expensive). And I think we'll only really ever want to do this with population data, so I don't know if we necessarily need a fully modular function?

hjohns12 commented:
> * Fixes the date mismatch issues (i.e. where we have `Residents.Population` data on 1/1/21 but run the scrapers on 1/2/21 and want to forward fill population data within a time window for a facility).

> 
> A few things to note:
> 
> * Both of these are parameters (`coalesce_pop` and `drop_na_covid`) that default to TRUE. Is that what we want? Are the names descriptive enough? Thought about renaming `coalesce` to `coalesce_facilities` but decided that might break things.
I see what you mean by naming it `drop_na_covid`, but I think it's a bit confusing. What if we renamed that parameter to something like `drop_noncovid_obs` or `drop_popOnly_obs` ? 

> * I'm using the same window parameter for the `all_dates = FALSE` forward-filling as historical population forward-filling. This feels okay to me, I think?
This means when I choose a window of 30, I'm saying: 
1. Give me data from the last 30 days
2. Forward-fill Residents.Population for, at most, 30 days
This seems ok to me too! Think the documentation for the `window` parameter needs to be updated to reflect this change, though. Separate issue, but I'm not sure if the existing documentation of the `window` parameter is accurate....

> * I decided to handle the date window here (as opposed to create a new version of `last_not_na` because it (1) felt like a cleaner solution (otherwise we'd have to pass around the Date vector and the Facility.ID vector to the function, which got messy with groupings, and (2) was MUCH faster (adding another looping nest over df rows was expensive). And I think we'll only really ever want to do this with population data, so I don't know if we necessarily need a fully modular function?
Makes sense to me!

hjohns12 reviewed (changes requested):
a few small tweaks, see previous comment!

nmmarquez reviewed (changes requested):
I feel like the window parameter here has two different functions and should be two different parameters. It fills in last observed data but only all_dates is false except for one variable which it fills in all values in between observations as long as that gap isn't more than window days. Its kind of confusing and think it would be better with two different parameters.

erika-tyagi commented:
All very good points!!  
* Created a separate `window_pop` param and added detail to the docs about this and the `window` param 
* Renamed `drop_na_covid ` to `drop_noncovid_obs`
* Updated parts of the documentation that seemed to be relics of copy and paste jobs 👀

hjohns12 reviewed (approved):
Looks great! Thank you!

nmmarquez reviewed (approved):
sweet!
